modules/javafx.graphics/src/test/java/test/javafx/application/ThemeTest.java

```diff
@@ -0,0 +1,68 @@
+package test.javafx.application;
+
+import javafx.application.Application;
+import javafx.application.Theme;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import org.junit.Test;
+import test.com.sun.javafx.pgstub.StubToolkit;
+
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+public class ThemeTest {
+
+    public static class PublicConstructor implements Theme {
```

Comment:
Rather than support varying constructor types to pass in the properties map, why not just specify the API as that platformThemeChanged will be called immediately after construction? That way there's only way code path that handles the properties map, and the way the class is constructed no longer matters.